components/attention/attentionUtil.js

```diff
@@ -30,3 +33,28 @@ export const computeCalloutArrow = ({ actualDirection, directionName, arrowEl })
   arrowEl.value.$el.style.left = directionIsVertical ? middlePosition : null
   arrowEl.value.$el.style.top = !directionIsVertical ? middlePosition : null
 }
+
+const variantClasses = {
+  callout: {
+    wrapper: ccAttention.callout,
+    arrow: ccAttention.arrowCallout
+  }, 
+  highlight: {
+    wrapper: ccAttention.highlight,
+    arrow: ccAttention.arrowHighlight
+  }, 
+  tooltip: {
+    wrapper: ccAttention.tooltip,
+    arrow: ccAttention.arrowTooltip
+  },
+  popover: {
+    wrapper: ccAttention.popover,
+    arrow: ccAttention.arrowPopover
+  }, 
+}
+
+export const getVariantClasses = (variantProps) => {
+  const variant = Object.keys(variantClasses).find(b => !!variantProps[b]) || '';
+
+  return variantClasses[variant];
+};
```

Comment:
Is there a reason (that I fail to see) for this function, and to not just export variantClasses? 🤔 🤷 😄

Reply:
Are the variant props supposed to be mutually exclusive? And you must supply at least one?

Reply:
> Are the variant props supposed to be mutually exclusive? And you must supply at least one?
You must supply at least one variant + there shouldn't be more than 1 variant applied. That calls for changing the API to having 1 `variant` prop that can have 1 of 4 values ('tooltip', 'callout', 'popover' or 'highlight'), but we didn't do that because we had decided not to introduce too many breaking changes when migrating from Fabric.

> Is there a reason (that I fail to see) for this function, and to not just export variantClasses?
It could be that in Vue there's no need for this function actually, and I could just add this to `computed()`: `[ccAttention.highlight]: props.highlight`. I'll look into that! In the case of React and Elements I had to return just the classes that are needed for given variant because `classnames` functions we used there would remove duplicate keys (class strings) - in our case that would be valid for `ccAttention.arrowHighlight` and `ccAttention.arrowCallout`, as they have the same strings of classes.

Reply:
> It could be that in Vue there's no need for this function actually, and I could just add this to computed(): `[ccAttention.highlight]: props.highlight`. I'll look into that!

Unfortunately, the same issue happens in computed(), so callout or highlight won't get the arrow styled properly, depending which class is defined first in the computed() statement. See example below:

```
  const arrowClasses = computed(() => ({
    [ccAttention.arrowBase]: true,
    [ccAttention[arrowDirectionClassname.value]]: true,  
    [ccAttention.arrowTooltip]: props.tooltip,
    [ccAttention.arrowCallout]: props.callout, // 'i-bg-$color-callout-background i-border-$color-callout-border'
    [ccAttention.arrowPopover]: props.popover,
    [ccAttention.arrowHighlight]: props.highlight // 'i-bg-$color-callout-background i-border-$color-callout-border'
}));
```
![Screenshot 2023-10-26 at 13 47 25](https://github.com/warp-ds/vue/assets/41303231/25080579-e0ea-45b0-b0cc-b47b59ab59ea)